Add methods to sign and recover messages/signatures to AccountManager

0x-r4bbit opened the pull request:
Also, make AccountManager a dependency of `Messenger`. This is needed for community token permissions as we'll need a way to access wallet accounts and sign messages when sending requests to join a community.

The APIs have been mostly taken from `GethStatusBackend` and `personal` service.

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (9)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 67bb2826 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3272/1/) | 2023-03-10 12:25:25 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230310-122321-67bb28-pr3272.zip) |
| :heavy_check_mark: | 67bb2826 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3272/1/) | 2023-03-10 12:30:09 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230310-122321-67bb28-pr3272.aar) |
| :heavy_multiplication_x: | 67bb2826 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/1/) | 2023-03-10 12:49:53 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/1//consoleText) |
| :heavy_check_mark: | 67bb2826 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3272/1/) | 2023-03-10 13:11:41 | ~48 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230310-122321-67bb28-pr3272.zip) |
| :heavy_multiplication_x: | 67bb2826 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/2/) | 2023-03-10 13:24:25 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 22adbb35 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3272/2/) | 2023-03-15 10:38:38 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230315-103647-22adbb-pr3272.zip) |
| :heavy_multiplication_x: | 22adbb35 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/3/) | 2023-03-15 10:39:32 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/3//consoleText) |
| :heavy_check_mark: | 22adbb35 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3272/2/) | 2023-03-15 10:40:22 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230315-103647-22adbb-pr3272.aar) |
| :heavy_check_mark: | 22adbb35 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3272/2/) | 2023-03-15 10:40:40 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230315-103647-22adbb-pr3272.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | df103525 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3272/3/) | 2023-03-15 11:21:35 | ~51 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230315-112043-df1035-pr3272.zip) |
| :heavy_check_mark: | df103525 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3272/3/) | 2023-03-15 11:24:09 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230315-112043-df1035-pr3272.zip) |
| :heavy_check_mark: | df103525 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3272/3/) | 2023-03-15 11:25:16 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230315-112043-df1035-pr3272.aar) |
| :heavy_check_mark: | df103525 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/4/) | 2023-03-15 11:32:44 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e89d3023 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3272/4/) | 2023-03-16 08:09:32 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230316-080830-e89d30-pr3272.zip) |
| :heavy_check_mark: | e89d3023 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3272/4/) | 2023-03-16 08:12:00 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230316-080822-e89d30-pr3272.aar) |
| :heavy_check_mark: | e89d3023 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3272/4/) | 2023-03-16 08:12:30 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230316-080825-e89d30-pr3272.zip) |
| :heavy_multiplication_x: | e89d3023 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/5/) | 2023-03-16 08:12:50 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/5//consoleText) |
| :heavy_multiplication_x: | e89d3023 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/6/) | 2023-03-16 08:55:46 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/6//consoleText) |
| :heavy_check_mark: | e89d3023 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/7/) | 2023-03-16 09:34:52 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3272/7//consoleText) |

0x-r4bbit committed:
Add methods to sign and recover messages/signatures to AccountManager

Also, make AccountManager a dependency of Messenger.
This is needed for community token permissions as we'll need a way to access wallet accounts
and sign messages when sending requests to join a community.

The APIs have been mostly taken from GethStatusBackend and personal service.

MishkaRogachev commented on account/accounts.go:
```diff
@@ -44,11 +51,24 @@ func (e ErrCannotLocateKeyFile) Error() string {
+type SignParams struct {
+	Data     interface{} `json:"data"`
+	Address  string      `json:"account"`
+	Password string      `json:"password"`
```
just curious is this a real password or just a password hash?

0x-r4bbit replied:
This is a password hash

MishkaRogachev commented on account/accounts.go:
```diff
@@ -612,3 +632,96 @@ func (m *Manager) ReEncryptKeyStoreDir(keyDirPath, oldPass, newPass string) erro
 	return m.keystore.Delete(types.Account{Address: address}, password)
 }
+
+func (m *Manager) GetVerifiedWalletAccount(db *accounts.Database, address, password string) (*SelectedExtKey, error) {
```
address types.Address

0x-r4bbit replied:
As you see this takes a string. It's exactly the same API that we have in wallet. I copied it over so we have it avaiable in messenger. unfortunately we can't easily depend on wallet in messenger due to how stuff is orchestrated at the moment. I'd rather not touch/change these functions

MishkaRogachev commented on account/accounts.go:
```diff
@@ -612,3 +632,96 @@ func (m *Manager) ReEncryptKeyStoreDir(keyDirPath, oldPass, newPass string) erro
+	}, nil
+}
+
+func (m *Manager) generatePartialAccountKey(db *accounts.Database, address string, password string) (*types.Key, error) {
```
types.Address?

0x-r4bbit replied:
Same here as https://github.com/status-im/status-go/pull/3272/files#r1138423666

MishkaRogachev reviewed (approved):
Looks good for me!
